### PR TITLE
Port shape estimation to .auto

### PR DESCRIPTION
**Pull request by wep21**
## Description

.auto porting for shape estimation

## Pre-Review Checklist for the PR Author

**PR Author should check the checkboxes below when creating the PR.**

- [x] Read [pull request guidelines][pull-request-guidelines]
- [x] Code follows [coding guidelines][coding-guidelines]
- [x] Assign PR to reviewer

## Checklist for the PR Reviewer

**Reviewers should check the checkboxes below before approval.**

- [ ] Commits are properly organized and messages are according to the guideline
- [ ] Code follows [coding guidelines][coding-guidelines]
- [ ] (Optional) Unit tests have been written for new behavior
- [ ] PR title describes the changes

## Post-Review Checklist for the PR Author

**PR Author should check the checkboxes below before merging.**

- [ ] All open points are addressed and tracked via issues or tickets
- [ ] Write [release notes][release-notes]

## CI Checks

- **Build and test for PR / build-and-test-pr**: Required to pass before the merge.
- **Build and test for PR / clang-tidy-pr**: NOT required to pass before the merge. It is up to the reviewer(s). Found false positives? See the [guidelines][clang-tidy-guidelines].
- **Check spelling**: NOT required to pass before the merge. It is up to the reviewer(s). See [here][spell-check-dict] if you want to add some words to the spell check dictionary.

[clang-tidy-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/ClangTidyGuideline/
[coding-guidelines]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/1194394777/T4
[pull-request-guidelines]: https://tier4.github.io/autoware.proj/tree/main/developer_guide/PullRequestGuideline/
[release-notes]: https://tier4.atlassian.net/wiki/spaces/AIP/pages/563774416
[spell-check-dict]: https://github.com/tier4/autoware-spell-check-dict#how-to-contribute

### CODE REVIEW

File: perception/object_recognition/detection/shape_estimation/src/node.cpp [hunk 1]
```diff
@@ -16,23 +16,25 @@
...
+
 #include <tf2/LinearMath/Matrix3x3.h>
 #include <tf2/LinearMath/Quaternion.h>
 #include <tf2/utils.h>
 #include <tf2_geometry_msgs/tf2_geometry_msgs.h>
 
 #include <memory>
 
-using SemanticType = autoware_perception_msgs::msg::Semantic;
+using SemanticType = autoware_auto_perception_msgs::msg::ObjectClassification;
```
Comment (yukkysaito) [inline perception/object_recognition/detection/shape_estimation/src/node.cpp:28]: SemanticTypeより今回はLabel or ClassLabelのほうが良いかもです

Comment (wep21) [reply perception/object_recognition/detection/shape_estimation/src/node.cpp]: fixed at https://github.com/tier4/AutowareArchitectureProposal.iv/pull/551/commits/62c15e1d30c25ff701a219cd880f25012bc127a4

File: perception/object_recognition/detection/shape_estimation/lib/shape_estimator.cpp [hunk 2]
```diff
@@ -60,21 +60,21 @@ bool ShapeEstimator::estimateShapeAndPose(
 
 bool ShapeEstimator::estimateShape(
   const int type, const pcl::PointCloud<pcl::PointXYZ> & cluster,
-  const boost::optional<float> & yaw, autoware_perception_msgs::msg::Shape & shape_output,
+  const boost::optional<float> & yaw, autoware_auto_perception_msgs::msg::Shape & shape_output,
   geometry_msgs::msg::Pose & pose_output)
 {
+  using autoware_auto_perception_msgs::msg::ObjectClassification;
```
Comment (yukkysaito) [inline perception/object_recognition/detection/shape_estimation/lib/shape_estimator.cpp:66]: Labelで統一しても良いかもです

File: perception/object_recognition/detection/shape_estimation/lib/shape_estimator.cpp [hunk 3]
```diff
@@ -84,15 +84,16 @@ bool ShapeEstimator::estimateShape(
 }
 
 bool ShapeEstimator::applyFilter(
-  const int type, const autoware_perception_msgs::msg::Shape & shape_output,
+  const int type, const autoware_auto_perception_msgs::msg::Shape & shape_output,
   const geometry_msgs::msg::Pose & pose_output)
 {
+  using autoware_auto_perception_msgs::msg::ObjectClassification;
```
Comment (yukkysaito) [inline perception/object_recognition/detection/shape_estimation/lib/shape_estimator.cpp:90]: Label

File: perception/object_recognition/detection/shape_estimation/lib/shape_estimator.cpp [hunk 4]
```diff
@@ -102,15 +103,16 @@ bool ShapeEstimator::applyFilter(
 }
 
 bool ShapeEstimator::applyCorrector(
-  const int type, const bool use_reference_yaw, autoware_perception_msgs::msg::Shape & shape_output,
-  geometry_msgs::msg::Pose & pose_output)
+  const int type, const bool use_reference_yaw,
+  autoware_auto_perception_msgs::msg::Shape & shape_output, geometry_msgs::msg::Pose & pose_output)
 {
+  using autoware_auto_perception_msgs::msg::ObjectClassification;
```
Comment (yukkysaito) [inline perception/object_recognition/detection/shape_estimation/lib/shape_estimator.cpp:109]: Label

File: perception/object_recognition/detection/shape_estimation/lib/shape_estimator.cpp (hunk 2 above)
Comment (wep21) [reply perception/object_recognition/detection/shape_estimation/lib/shape_estimator.cpp]: fixed at https://github.com/tier4/AutowareArchitectureProposal.iv/pull/551/commits/328b2d98a08daf49beb0a9f46f8c0cf96ba19a7c

File: perception/object_recognition/detection/shape_estimation/lib/shape_estimator.cpp (hunk 3 above)
Comment (wep21) [reply perception/object_recognition/detection/shape_estimation/lib/shape_estimator.cpp]: fixed at https://github.com/tier4/AutowareArchitectureProposal.iv/pull/551/commits/328b2d98a08daf49beb0a9f46f8c0cf96ba19a7c

File: perception/object_recognition/detection/shape_estimation/lib/shape_estimator.cpp (hunk 4 above)
Comment (wep21) [reply perception/object_recognition/detection/shape_estimation/lib/shape_estimator.cpp]: fixed at https://github.com/tier4/AutowareArchitectureProposal.iv/pull/551/commits/328b2d98a08daf49beb0a9f46f8c0cf96ba19a7c

**Comment by yukkysaito** [comment]
大変申し訳無いのですがspellcheckでConvexhullで引っかかるのでConvexHullに変えてもらってもいいでしょうか？ :pray:

**Comment by wep21** [comment]
@yukkysaito I fixed typo(Convexhull -> ConvexHull) at https://github.com/tier4/AutowareArchitectureProposal.iv/pull/551/commits/e251e92e52b6a7e82474f88aa79cc7ec378e94e9.

**Review by yukkysaito — APPROVED** [review]

File: perception/object_recognition/detection/shape_estimation/lib/shape_estimator.cpp [hunk 5]
```diff
@@ -60,39 +62,36 @@ bool ShapeEstimator::estimateShapeAndPose(
...
-    model_ptr.reset(new ConvexhullShapeModel());
+    model_ptr.reset(new ConvexHullShapeModel());
   }
 
   return model_ptr->estimate(cluster, shape_output, pose_output);
 }
 
 bool ShapeEstimator::applyFilter(
-  const int type, const autoware_perception_msgs::msg::Shape & shape_output,
+  const int type, const autoware_auto_perception_msgs::msg::Shape & shape_output,
```
Comment (yukkysaito) [inline perception/object_recognition/detection/shape_estimation/lib/shape_estimator.cpp:86]: typeという用語は使わないようになったので
const std::uint8_t label
でお願いします

File: perception/object_recognition/detection/shape_estimation/lib/shape_estimator.cpp [hunk 6]
```diff
@@ -102,15 +101,15 @@ bool ShapeEstimator::applyFilter(
 }
 
 bool ShapeEstimator::applyCorrector(
-  const int type, const bool use_reference_yaw, autoware_perception_msgs::msg::Shape & shape_output,
-  geometry_msgs::msg::Pose & pose_output)
+  const int type, const bool use_reference_yaw,
```
Comment (yukkysaito) [inline perception/object_recognition/detection/shape_estimation/lib/shape_estimator.cpp:104]: typeという用語は使わないようになったので
const std::uint8_t label
でお願いします

File: perception/object_recognition/detection/shape_estimation/src/node.cpp [hunk 7]
```diff
@@ -42,24 +44,23 @@ ShapeEstimationNode::ShapeEstimationNode(const rclcpp::NodeOptions & node_option
...
   // Create output msg
-  autoware_perception_msgs::msg::DynamicObjectWithFeatureArray output_msg;
+  autoware_auto_perception_msgs::msg::DetectedObjects output_msg;
   output_msg.header = input_msg->header;
 
   // Estimate shape for each object and pack msg
   for (const auto & feature_object : input_msg->feature_objects) {
     const auto & object = feature_object.object;
-    const auto & type = object.semantic.type;
+    const auto & type = object.classification.front().label;
```
Comment (yukkysaito) [inline perception/object_recognition/detection/shape_estimation/src/node.cpp:61]: label

**Review by yukkysaito — COMMENTED** [review]
何度もすいません

File: perception/object_recognition/detection/shape_estimation/lib/shape_estimator.cpp (hunk 6 above)
Comment (wep21) [reply perception/object_recognition/detection/shape_estimation/lib/shape_estimator.cpp]: https://github.com/tier4/AutowareArchitectureProposal.iv/pull/551/commits/f41cbc9f6bf0db394e05a2d0abe3af15db85b5a3

File: perception/object_recognition/detection/shape_estimation/src/node.cpp (hunk 7 above)
Comment (wep21) [reply perception/object_recognition/detection/shape_estimation/src/node.cpp]: fixed at https://github.com/tier4/AutowareArchitectureProposal.iv/pull/551/commits/f41cbc9f6bf0db394e05a2d0abe3af15db85b5a3

**Review by yukkysaito — APPROVED** [review]

**Review by yukkysaito — APPROVED** [review]

**Comment by wep21** [comment]
@yukkysaito Can I proceed to merge?

**Comment by kaancolak** [comment]
Hi everyone, I am planning port tier4's shape_estimation to Autoware.Auto. Releted Issue here : https://gitlab.com/autowarefoundation/autoware.auto/AutowareAuto/-/issues/1399
 I think you are already changed its message type to new Autoware.Auto message type. Does anyone working this porting? 
 Thanks.
 @wep21 @yukkysaito

**Comment by yukkysaito** [comment]
@kaancolak 
Thank you for comment.
This shape estimation was made to support the new type of Autoware.auto.
but autoware_auto_msgs have not been merged, and I ported under the assumption that these changes have been merged.
https://gitlab.com/autowarefoundation/autoware.auto/autoware_auto_msgs/-/merge_requests

We plan to deploy this repository to the AWF Autoware.Auto.Universe by the end of December.
If you have any questions, please contact me or @mitsudome-r

**Pull request merged**